src/System.Net.HttpListener/tests/SimpleHttpTests.cs

```diff
@@ -159,5 +165,66 @@ public async Task UnknownHeaders_Success(int numHeaders)
                 }
             }
         }
+
+        [Fact]
+        public void ListenerRestart_BeginGetContext_Success()
+        {
+            using (HttpListenerFactory factory = new HttpListenerFactory())
+            {
+                HttpListener  listener = factory.GetListener();
+                listener.BeginGetContext((f) => { }, null);
+                listener.Stop();
+                listener.Start();
+                listener.BeginGetContext((f) => { }, null);
+            }
+        }
+
+        [ConditionalFact]
+        public async Task ListenerRestart_GetContext_Success()
+        {
+            string content = "ListenerRestart_GetContext_Success";
+            using (HttpListenerFactory factory = new HttpListenerFactory())
+            using (HttpClient client = new HttpClient())
+            {
+                HttpListener  listener = factory.GetListener();
```

Comment:
Should `listener` be used or closed here?

Reply:
same as above